src/Articles/containers/CreateArticle.jsx

```diff
@@ -0,0 +1,279 @@
+  Form, Row, Col, Button,
+} from 'react-bootstrap';
+import React from 'react';
+import '../CSS/CreateArticle.scss';
+import { connect } from 'react-redux';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCamera } from '@fortawesome/free-solid-svg-icons';
+import Loader from 'react-loader';
+import { storage } from '../../commons/Util/FirebaseConfig';
+import postAnArticle from '../redux/actions/CreateArticleAction';
+import ErrorLabel from '../../commons/components/ErrorLabel';
+import SuccessLabel from '../../commons/components/SuccessLabel';
+import history from '../../commons/history';
+
+export class CreateArticle extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isFieldInvalid: {
+        title: false,
+        body: false,
+        tags: false,
+      },
+      articleTitle: '',
+      articleBody: '',
+      articleTags: [],
+      articlePhoto: {
+        name: 'No Photo selected',
+        data: null,
+      },
+    };
+
+    this.handleCreateArticle = this.handleCreateArticle.bind(this);
+
+    this.handleChange = this.handleChange.bind(this);
+
+    this.handleImageUpload = this.handleImageUpload.bind(this);
+  }
+
+  componentWillMount() {
+    const token = localStorage.getItem('user_token');
+    if (token === null) {
+      history.push('/login');
+    }
+
+    return true;
+  }
+
+  handleCreateArticle = (event) => {
+    event.preventDefault();
+    const {
+      articleTitle, articleBody, articleTags, articlePhoto,
+    } = this.state;
+    const tags = articleTags.split(',');
+    const articleData = {
+      article: {
+        title: articleTitle,
+        body: articleBody,
+        description: articleBody,
+        tagName: tags,
+      },
+    };
+
+    const { postAnArticle } = this.props;
+
+    if (articlePhoto.data !== null) {
+      this.handleImageUpload();
+    } else {
+      postAnArticle(articleData);
+    }
+  }
+
+  handleChange = (event) => {
+    const { target: { name, value, files } } = event;
+    const { isFieldInvalid } = this.state;
+
+    switch (name) {
+      case 'articleTitle':
+        isFieldInvalid.title = value.length < 5;
+        break;
+      case 'articleBody':
+        isFieldInvalid.body = value.length < 100;
+        break;
+      case 'articleTags':
+        isFieldInvalid.tags = value.length < 3;
+        break;
+      case 'articlePhoto':
+        this.setState({
+          articlePhoto: {
+            name: value,
+            data: files[0],
+          },
+        });
+        break;
+      default:
+        break;
+    }
+    if (name !== 'articlePhoto') {
+      this.setState({
+        [name]: value,
+      });
+    }
+  }
+
+  handleImageUpload = () => {
+    const {
+      articleTitle, articleBody, articleTags, articlePhoto,
+    } = this.state;
+    const tags = articleTags.split(',');
+    const imageUpLoad = storage.ref(`article-image/${articlePhoto.data.name}`).put(articlePhoto.data);
```

Comment:
Like the way you are concatenation the strings here

Reply:
@jennizalwango thanks.